### PR TITLE
feature-5232-Trivy-Security-Scans

### PR DESCRIPTION
**Pull request by btme0011**
## What does this PR do?

Added Trivy scan

## Test Plan

Ran the scan on my personal forked Repo [link](https://github.com/btme0011/appwrite/actions/runs/6469467182/job/17563582936)

## Related PRs and Issues

- #5232 
## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Comment by btme0011** [comment]
Currently the scan is failing Created an issue for it - [link](https://github.com/appwrite/appwrite/issues/6877)

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
```
Comment (stnguyen90) [inline .github/workflows/trivy.yml:15]:
```suggestion
```

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    - name: Build the Docker image
+      run: docker build . -t appwrite_image:latest
```
Comment (stnguyen90) [inline .github/workflows/trivy.yml:18]: Let's build like how we do in tests to make use of caching: 

https://github.com/appwrite/appwrite/blob/fce18647d2352e4cfce7ce75fd103a21bafbf7d0/.github/workflows/tests.yml#L28-L44

File: .github/workflows/trivy.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Trivy
```
Comment (stnguyen90) [inline .github/workflows/trivy.yml:1]: Let's name the file scan-container.yml and make this name Scan container. The workflow and goal is to have scans for containers and trivy is just a tool that may get swapped out later.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

**Comment by gewenyu99** [comment]
Hey there! There were a lot of big PRs during this Hacktoberfest, and we wanted to give everyone ample time to collaborate with our engineering team. If you were able to merge your PRs during October, amazing. If it’s still not merged, don’t worry about it either. Either way, we’ve got your Hacktoberfest swag minted and ready to ship. 

Please comment with your Discord username here so we can contact you about your shipping information to deliver your Hacktoberfest swag.

**Comment by EVDOG4LIFE** [comment]
Refactored this into a nightly running job that builds the image from source and runs a filesystem scan to scan for potential code vulnerabilities. 

Sample of what they look like - https://github.com/EVDOG4LIFE/appwrite_wfTest/security/code-scanning

FS and Image scan happen in parallel - see successful run on my fork here - https://github.com/EVDOG4LIFE/appwrite_wfTest/actions/runs/10206787738

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**